content/en/dev-reference/security/single-sign-on.md

```diff
@@ -487,3 +487,116 @@ If you decide to provide a custom mapping, here is the context of the script:
+
+- **`code`**: This parameter is the code received from the user's authorization process.
+- **`config`**: This is a JSON object that contains all the configuration settings defined for the identity provider, including `client_id`, `client_secret`, `scope`, `oauthRedirectUri` (which corresponds to the OAuth redirect URI), and `token_url` (which corresponds to the Access Token URL).
+- **`userRecord`**: This parameter represents the user being authenticated.
+
+As the first step, the script must obtain the `access_token` by making a call to the API defined by the provider, using the required parameters. This process may vary from provider to provider. Once the `access_token` is obtained, a call can be made to the provider's API to retrieve the user's data. The script should first check if the user already exists by searching for their email. If the user exists, return their record along with the associated firstName and lastName. If the user does not exist, the developer must ensure that the fields `firstName`, `lastName`, and `email` are filled out; otherwise, the user record cannot be saved, as these fields are required. Finally, the script must return the userRecord.
+
+##### Samples
+
+  ```js
+    // get access token
+  const accessTokenResponse = svc.http.post({
+    url: config.tokenUrl,
+    params: {
+      grant_type: 'authorization_code',
+      client_id: config.clientId,
+      redirect_uri: config.oauthRedirectUri,
+      code: code
+    },
+    headers: {
+      'Content-Type': 'application/json',
+    }
+  });
+  if (!accessTokenResponse || !accessTokenResponse.access_token) throw('Error obtaining access_token')
+  // get user data
+  const userDataResponse = svc.http.post({
+    url: 'https://shopify.com/87175397693/account/customer/api/2024-10/graphql',
+    params: {
+      query: 'query { customer { id firstName lastName emailAddress {emailAddress} } }'
+    },
+    headers: {
+      'Content-Type': 'application/json',
+      Authorization : accessTokenResponse.access_token
+    }
+  });
+  if (!userDataResponse || !userDataResponse.data || !userDataResponse.data.customer) throw('Error obtaining customer data')
+  // fill out userRecord
+  let email;
+  if (userDataResponse.data.customer.emailAddress) {
+    email = userDataResponse.data.customer.emailAddress.emailAddress;
+    if (email) {
+      const userRegistered = sys.data.findOne('sys.users', {email: email});
+      if (userRegistered) {
+        return userRegistered;
+      }
+    }
+  }
+  const firstName = userDataResponse.data.customer.firstName || 'Default first name';
+  const lastName = userDataResponse.data.customer.lastName || 'Default last name';
+  if (!email) throw('Missing required fields: email') // you might throw an exception for firstName and lastName
+  userRecord.field('firstName').val(firstName);
+  userRecord.field('lastName').val(lastName)
+  userRecord.field('email').val(email)
```

Comment:
I would add an optional field to indicate that it is a good time to update the user.
```suggestion
  userRecord.field('firstName').val(firstName);                              // Required
  userRecord.field('lastName').val(lastName);                              // Required
  userRecord.field('email').val(email);                                           // Required
  userRecord.field('currentlyWorkingOn').val("Google");              // Optional
```